File: 3075. Maximize Happiness of Selected Children.java
class Solution {
    public long maximumHappinessSum(int[] happiness, int k) {
        Arrays.sort(happiness);
        int n = happiness.length;
        long answer = 0;
        for (int i = n-1; i >= 0 && k > 0; i--, k--) {
            answer += Math.max(happiness[i] - n + i + 1, 0);
        }
        return answer;
    }
}

File: 3075. Maximize Happiness of Selected Children.go
func maximumHappinessSum(happiness []int, k int) int64 {
	sort.Slice(happiness, func(i, j int) bool {
		return happiness[i] > happiness[j]
	})
	var answer int64
	for i := 0; i < k; i++ {
		if happiness[i] > i {
			answer += int64(happiness[i] - i)
		}
	}
	return answer
}